op-service/sources/l1_beacon_client.go

```diff
@@ -279,35 +279,42 @@ func (cl *L1BeaconClient) GetBlobs(ctx context.Context, ref eth.L1BlockRef, hash
 	return blobs, nil
 }
 
+// blobsFromSidecars fetches blobs from the given blob sidecars, verifiy the blob data against the hash of the commitment
+// (if skipBlobVerification is false).
 func blobsFromSidecars(blobSidecars []*eth.BlobSidecar, hashes []eth.IndexedBlobHash, skipBlobVerification bool) ([]*eth.Blob, error) {
 	if len(blobSidecars) != len(hashes) {
 		return nil, fmt.Errorf("number of hashes and blobSidecars mismatch, %d != %d", len(hashes), len(blobSidecars))
 	}
-
 	out := make([]*eth.Blob, len(hashes))
 	for i, ih := range hashes {
 		sidecar := blobSidecars[i]
 		if sidx := uint64(sidecar.Index); sidx != ih.Index {
 			return nil, fmt.Errorf("expected sidecars to be ordered by hashes, but got %d != %d", sidx, ih.Index)
 		}
-
-		// make sure the blob's kzg commitment hashes to the expected value
-		hash := eth.KZGToVersionedHash(kzg4844.Commitment(sidecar.KZGCommitment))
-		if hash != ih.Hash {
-			return nil, fmt.Errorf("expected hash %s for blob at index %d but got %s", ih.Hash, ih.Index, hash)
-		}
-
 		if !skipBlobVerification {
-			// confirm blob data is valid by verifying its proof against the commitment
-			if err := eth.VerifyBlobProof(&sidecar.Blob, kzg4844.Commitment(sidecar.KZGCommitment), kzg4844.Proof(sidecar.KZGProof)); err != nil {
-				return nil, fmt.Errorf("blob at index %d failed verification: %w", i, err)
+			if err := verifyBlobProof(&sidecar.Blob, ih.Hash); err != nil {
+				return nil, fmt.Errorf("blob %d failed verification: %w", i, err)
 			}
 		}
 		out[i] = &sidecar.Blob
 	}
 	return out, nil
 }
 
+// verifyBlobProof verifies that the blob data corresponds to the provided commitment.
+// It recomputes the commitment from the blob data and checks it matches the expected commitment hash.
+func verifyBlobProof(blob *eth.Blob, expectedCommitmentHash common.Hash) error {
```

Comment:
technically, we're verifying the commitment hash, not the proof

```suggestion
func verifyBlobCommitmentHash(blob *eth.Blob, expectedCommitmentHash common.Hash) error {
```

Reply:
[6548dae](https://github.com/ethereum-optimism/optimism/pull/17725/commits/6548dae11e89428948c718c9845065457e327085)